src/NLog/Internal/StringBuilderExt.cs

```diff
@@ -192,11 +192,22 @@ public static void AppendXmlDateTimeRoundTrip(this StringBuilder builder, DateTi
         /// <param name="builder"></param>
         public static void ClearBuilder(this StringBuilder builder)
         {
+            try
+            {
 #if !SILVERLIGHT && !NET3_5
-            builder.Clear();
+                builder.Clear();
 #else
-            builder.Length = 0;
+                builder.Length = 0;
 #endif
+            }
+            catch
+            {
+                // Default StringBuilder Clear() can cause the StringBuilder to re-allocate new internal char-array
```

Comment:
where could you see this behavior?

Reply:
https://referencesource.microsoft.com/#mscorlib/system/text/stringbuilder.cs,498 <-- When doing `Length = 0` then it will drop all chained StringBuilder-buffers and instead allocate a single char-buffer with total capacity. This behavior doesn't occur when doing `Remove(0, Length - 1)`.